_posts/2022-04-27-newpipe-0.23.0.md

```diff
@@ -0,0 +1,176 @@
+
+- [@Stypox](https://github.com/Stypox) added the Checkstyle static code analysis tool to the NewPipe Extractor project, so that code formatting remains standardised throughout the repo. A bunch of code was also refactored along the way. (The Checkstyle mentioned in previous blog posts is part of the NewPipe repo.)
+
+- [@TacoTheDank](https://github.com/TacoTheDank) replaced the `CircleImageView` library with Google's own Material library, `ShapeableImageView`. Taco dares you to find the difference between the old and new icons. (Hint for those who really want to try: look at circular image icons, like the channel avatars.) Aaanyway, that's one less dependency! Less code is good code.
+
+- [@Stypox](https://github.com/Stypox) made a lot of complicated changes that we don't understand at all so that there is no raw use of parameterized classes in the code, which generates a Java warning. We're assuming this is a good thing. Let us know in the comments.
+
+- [@Stypox](https://github.com/Stypox) fixed another build warning, this time by AndroidX, about database queries returning unused columns. For this, AndroidX's `RewriteQueriesToDropUnusedColumns` function was used.
+
+- [@Atemu](https://github.com/Atemu) fixed the app failing to build on Macbooks with Apple's in-house (M1, etc.) chips by updating the AndroidX Room version.
+
+- [@TobiGr](https://github.com/TobiGr) refactored the way `InfoItemDialog`s are generated, as the previous code was not up to par in terms of legibility, structure, consistency, code duplication, and separation of concerns.
+
+- [@avently](https://github.com/avently) implemented a better equality check for queues, which ensures that going up the backstack via the Back button won't result in duplicates.
+
+- [@TacoTheDank](https://github.com/TacoTheDank), in their LibraryMan avatar:
+  - Gradle (and wrapper): 7.3.3 -> 7.4.2
+  - Android Gradle Plugin: 7.0.4 -> 7.1.3
+  - buildTools: 30.0.3 -> 31.0.0
+  - Kotlin: 1.5.31 -> 1.6.20
+  - ktlint: 0.43.2 -> 0.44.0
+  - Checkstyle: 9.2.1 -> 9.3
+  - Google Material: 1.4.0 -> 1.5.0 (which allowed one method to be updated to a much simpler version)
+  - AndroidX:
+    - LocalBroadcastManager: 1.0.0 -> 1.1.0
+    - Media: 1.4.3 -> 1.5.0
+    - Preference: 1.1.1 -> 1.2.0
+    - ConstraintLayout: 2.1.2 -> 2.1.3
+    - Room: 2.4.1 -> 2.4.2
+  - GitHub Actions dependencies:
+    - github-script: v5 -> v6
+    - and all the rest: v2 -> v3
+
+- While investigating some player UI layout issues, [@Stypox](https://github.com/Stypox) realised that the `large-land` XML layout file wasn't actually being used at all, and removed it, getting rid of some 800 lines of duplicated code!
+
+- [@TacoTheDank](https://github.com/TacoTheDank) migrated the app update checker from using the deprecated `IntentService` service to using a worker, by adding the AndroidX `Work` library. There was also some Kotlin-isation and refactoring involved.
```

Comment:
Probably just a nitpick of mine lol, but I would change this up a little:

``
[@TacoTheDank](https://github.com/TacoTheDank) migrated the app update checker from using the deprecated `IntentService` service to using a worker (courtesy of the AndroidX Work library). This fixes the issue mentioned earlier where the update check would randomly fail while the app ran in the background, causing the app to crash. There was also some Kotlin-isation and refactoring involved.
``

Reply:
```suggestion
- [@TacoTheDank](https://github.com/TacoTheDank) migrated the app update checker from using the deprecated `IntentService` service to using a worker, by adding the AndroidX `Work` library. There was also some Kotlin-isation and refactoring involved. Who doesn't like themselves some Kotlin.
```